Relationship/SpearframeRelationship.py

```diff
@@ -27,20 +27,49 @@ def __init__(self, sensor_x, sensor_y):
         # set up our parent class
         Relationship.__init__(self, sensor_x, sensor_y)
 
+        # sensor_x
+        self.x_mono_list = []
+
+        # sensor_y
+        self.y_mono_list = []
+
     def __should_generate_new_frame(self, x_vals, y_vals):
 
         # Make sure their equal in length before going further
-        if len(x_vals) != len(y_vals):
+        if len(x_vals) != len(y_vals) and len(x_vals) < 3 or len(y_vals) < 3:
```

Comment:
I think it should actually be

```python
len(x_vals) != len(y_vals) or len(x_vals) < 3
```

shouldn't generate whenever the list are unequal, or when one of the items is less than 3.
If their both equal then it doens't matter which one we check of it's length for 3